### DISCUSSION TITLE
Extract value from a file generated in a process to a value in the channel?

### DISCUSSION THREAD

**Discussion by davidmasp** [nextflow-io/nextflow#2100] [Q&A] [answered]
I am sure this is super easy but I am a bit stuck. I am trying to extract the value of a file that I generate in a process to put it directly in a channel as a value. 

So far I am trying this but not working

```
cheers = Channel.from 'Bonjour', 'Ciao', 'Hello', 'Hola'

process sayHello {
  echo false
  input:
    val x from cheers
  output:
    val file("out.tst").getText() into out_channel
  script:
    """
    echo '$x world!' > out.tst
    """
}

process test {
  echo true
  input:
    val x from out_channel
  script:
    """
    echo ${x}
    """
}

```

**Comment by davidmasp** [comment, marked as answer]
Okay, I made it work I think. Not sure if it's the most elegant way though.

```
#!/usr/bin/env nextflow

cheers = Channel.from 'Bonjour', 'Ciao', 'Hello', 'Hola'

process sayHello {
  echo false
  input:
    val x from cheers
  output:
    file "${x}_out.tst" into out_channel
  script:
    """
    echo '$x world!' > ${x}_out.tst
    """
}

out_channel.map{it.getText()}.set{out_channel2}




process test {
  echo true
  input:
    val x from out_channel2
  script:
    """
    echo ${x}
    """
}
```

**Reply by pditommaso** [reply]
That's the way to go. In alternative, you can use an env variable, but it's very similar under the hood https://www.nextflow.io/docs/latest/process.html#output-env

**Reply by davidmasp** [reply]
oh nice! that's more or less what I was looking for initially. 

For my future record, I ended up using this `.readLines().join()`  for files with one line to avoid the new line character

**Reply by albertoriva** [reply]
> That's the way to go. In alternative, you can use an env variable

I have a related question: what if I want to capture *two* env variables, into a tuple? I tried this:

```
output:
tuple env A, env B into output_ch
```

But the script did not compile. Is there a way to do this?
